### PR TITLE
Setup Mock Data

### LINKED ISSUE #10: Define API types
**Type**: full stack
**FOF**:  3

## Description
- Define types for various API requests/responses that are used populate election/candidate info
- Use those defined types to write a script that populates mock data that will be used for testing

### Step 1 - Read endpoint documentation to understand the structure of the API request/response
Example: https://developers.civicengine.com/docs/api/v1/get-elections

### Step 2 - Define Type for API Request/Response
image
image

## Todo

- [x] [Civic Engine - Elections](https://developers.civicengine.com/docs/api/v1/get-elections)
- [x] [Civic Engine - Positions](https://developers.civicengine.com/docs/api/v1/get-positions)
- [ ] [Google Civic](https://developers.google.com/civic-information/docs/v2)
- [ ] [Democracy Works](https://www.democracy.works/elections-api/)

## Learning Objectives
- TypeScript
- API's

### LINKED ISSUE #15: Populate mock data
**Type**: full stack
**FOF**:  2

## Description
- Use those defined types from #10 to write a script that populates mock data that will be used for testing
 
### Step 1 - Write script using [faker-js](https://github.com/faker-js/faker) to populate mock data
image

### Step 2 - Generate and validate mock data
```javascript
node generate-api.js
```
image

## Todo

- [x] [Civic Engine - Elections](https://developers.civicengine.com/docs/api/v1/get-elections)
- [ ] [Civic Engine - Positions](https://developers.civicengine.com/docs/api/v1/get-positions)
- [ ] [Google Civic](https://developers.google.com/civic-information/docs/v2)
- [ ] [Democracy Works](https://www.democracy.works/elections-api/)

## Learning Objectives
- Faker-JS
- Data Infrastructure

### LINKED ISSUE #16: Migrate workflows to Node 16
**Type**: devops
**FOF**:  1

## Description

Update the node version on the client and api workflows to node 16 instead of node 12

## Todo

- [ ] Migrate Client workflow
- [ ] Migrate API workflow

### LINKED ISSUE #22: 🪗 Accordion component
**Type**: client
**FOF**:  2

## Description
Implement and style a generic accordion component

## Screenshot
image

## Todo

- [ ] Create a folder that stores generic UI components
- [ ] Implement the skeleton of the Accordion component
- [ ] Style the accordion to match the Figma

## Links
- [Chakra UI - Accordion](https://chakra-ui.com/docs/disclosure/accordion)
- [Figma](https://www.figma.com/file/6aJ3cfu2Yq5dxUQcNDkm6Y/Brink-Designs?node-id=34%3A2)

### PR DESCRIPTION
**Pull request by amit-sawhney**
## Status:
:rocket: Ready

## Summary

Setup types for `/elections` and `/positions` and generate mock data
Closes #10, Closes #15, Closes #16

## Changes

- Add Civic Engine-Elections
- Add Civic Engine-Positions
- Change Node version from 12 to 16
- Generate mock data for Elections and Position

### CODE REVIEW

File: client/data/script/generate-api.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    },
+    () => faker.name.findName(),
+  );
+
+  const generateCivicEnginePosition = () => ({
+    position_id: faker.datatype.number({
+      min: 1,
+      max: 100000,
+    }),
+    paperwork_instructions: faker.lorem.paragraph(3),
```
Comment (ayanmallik) [inline client/data/script/generate-api.js:70]: Nice job using the paragraph() function to populate longer fields

File: client/data/script/generate-api.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        'city',
+        'local',
+        'party',
+      ]),
+    },
+    state: faker.address.stateAbbr(),
+    candidates: candidate_names,
+    description: faker.lorem.paragraph(3),
+    name: faker.name.jobTitle(),
+    level: faker.random.arrayElement([
```
Comment (ayanmallik) [inline client/data/script/generate-api.js:100]: Good idea treating this field like an enum, we can specify the possible fields more in depth after we confirm the usage of this API

**Review by ayanmallik — APPROVED** [review]
Great job leveraging the faker() library to populate mock data for the /positions endpoint

File: client/src/types/civicEngine/positions.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  address: string; //350+5th+New+York+NY+10118
+  coords: {
+    latitude: number;
+    longitude: number;
+  };
+  page_size: number;
+  result_count: number;
+}
+
+export interface CivilEnginePosition {
```
Comment (amit-sawhney) [inline client/src/types/civicEngine/positions.ts:14]: Typo

**Review by amit-sawhney — COMMENTED** [review]
Feel free to merge after typo fix

**Pull request merged**